.github/workflows/ai-validation.yml

```
@@ -36,16 +36,20 @@ jobs:
         uses: actions/setup-node@v4
         with:
-          node-version: '22.14.0'
+          node-version: ${{ fromJson(steps.package.outputs.packageJson).engines.node }}
           cache: 'pnpm'
 
       - name: Install pnpm
-        uses: pnpm/action-setup@v2
+        uses: pnpm/action-setup@v4
         with:
-          version: 10.10.0
+          version: ${{ (fromJson(steps.package.outputs.packageJson).packageManager || '').replace('pnpm@', '') }}
```

actionlint (1.7.4)

parser did not reach end of input after parsing the expression. 5 remaining token(s) in the input: "(", "STRING", ",", "STRING", ")"

(expression)